packages/button/src/LionButton.js

```diff
@@ -172,12 +179,22 @@ export class LionButton extends DisabledWithTabIndexMixin(SlotMixin(LitElement))
   }
 
   /**
-   * Dispatch submit event and invoke submit on the native form when clicked
+   * Delegate click, by flashing a native button as a direct child
+   * of the form, and firing click on this button. This will fire the form submit
+   * without side effects caused by the click bubbling back up to lion-button.
    */
-  __clickDelegationHandler() {
-    if (this.type === 'submit' && this._nativeButtonNode && this._nativeButtonNode.form) {
-      this._nativeButtonNode.form.dispatchEvent(new Event('submit'));
-      this._nativeButtonNode.form.submit();
+  __clickDelegationHandler(e) {
+    if (this.constructor.__isIE11()) {
+      e.stopPropagation();
+    }
+
+    if (this.type === 'submit' && e.target === this) {
```

Comment:
Maybe it could be a good idea to also add a check for the type `reset`, something like
```javascript
if((this.type === 'submit' || this.type === 'reset') && e.target === this) { // Here you check if it is reset type
   if (this._form) {
        const nativeButton = document.createElement('button');
        nativeButton.type = this.type; // The button is created with the correct type
        this._form.appendChild(nativeButton);
        nativeButton.click();
        this._form.removeChild(nativeButton);
      }
    }
}
```
This way `lion-button` will also be able to clear all the inputs inside a form

Reply:
Aah yeah good catch, we supported that (but did not test) it in the past so we should support it now. I added a commit on top for it and included a test in it.